### PR TITLE
add cache module

### PR DESCRIPTION
**Pull request by ZhongliGao**

### CODE REVIEW

File: src/cache/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "cache"
+version = "0.1.0"
+edition = "2021"
+publish = false
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+linked-hash-map = "0.5"
```
Comment (PsiACE) [inline src/cache/Cargo.toml:10]: The maintenance state of linked-hash-map is not good, you can consider using ritelinked to repalce it.

hashlink is also possible, but I think ritelinked would be better.

Comment (ZhongliGao) [reply src/cache/Cargo.toml:10]: Thank you for your suggestion. I'll consider replacing it later.

Comment (ZhongliGao) [reply src/cache/Cargo.toml:10]: > The maintenance state of linked-hash-map is not good, you can consider using ritelinked to repalce it.
> 
> hashlink is also possible, but I think ritelinked would be better.

Thank you for your suggestion. I'll consider replacing it later.

**Comment by huachaohuang** [comment]
@ZhongliGao Thanks for your contribution! You can consider converting this PR to a draft before it is ready. It is also more friendly to provide some description about the PR.

**Comment by ZhongliGao** [comment]
> @ZhongliGao Thanks for your contribution! You can consider converting this PR to a draft before it is ready. It is also more friendly to provide some description about the PR.

**Comment by ZhongliGao** [comment]
> @ZhongliGao Thanks for your contribution! You can consider converting this PR to a draft before it is ready. It is also more friendly to provide some description about the PR.

Ok

**Pull request closed without merging**